int32 width
int32 height
bool[] data